### DISCUSSION TITLE
Is it possible to make a custom-drawn chart area?

### DISCUSSION THREAD

**Discussion by marklam** [oxyplot/oxyplot-avalonia#51] [Q&A] [answered]
Currently I have a custom heatmap-style Avalonia control which overrides the Render method to call `SKCanvas.DrawPaint` on the underlying `SKCanvas`. It's hosted in a `ZoomBorder` from Avalonia.Controls.PanAndZoom for interaction.

I would like to have OxyPlot axes (and maybe even  tooltips/trackers) but keep the ability to draw the plot area myself (because it involves shader effects). Is there a way to do this (or similar) with OxyPlot.Avalonia?

Thanks for your time
Mark.

**Comment by VisualMelon** [comment, marked as answer]
I don't think I've understood what you mean; there is an Overlay example which may or may not be useful: shows how to listen to the axes to update a component that isn't managed by OxyPlot.

Another thing you could do is just grab the canvas that OxyPlot draws to and stuff whatever you want in the back of that as part of a custom annotation (so that it gets access to the usual APIs that annotations use) by passing the `IRenderContext` interface (i.e. use reflection to get at the innards of the concrete `CanvasRenderContext`)

Overlay example: https://github.com/oxyplot/oxyplot-avalonia/blob/master/Source/Examples/Avalonia/AvaloniaExamples/Examples/OverlayDemo/MainWindow.xaml.cs
`CreateAndAdd` in `CanvasRenderContext`: https://github.com/oxyplot/oxyplot-avalonia/blob/master/Source/OxyPlot.Avalonia/CanvasRenderContext.cs#L866-L885 will give you an element on the canvas

In any case, this isn't really supported by the library, but happy to help bodge something.

**Reply by marklam** [reply]
Thanks! that first example looks like I can build on that. (Sorry for the slow response.)
